### DISCUSSION TITLE
random error

### DISCUSSION THREAD

**Discussion by rorykilpatrick** [ImageMagick/ImageMagick#2208] [Help]
version 7.17 trying to convert a large multipage pdf to png we randomly once a day get the following error 

"System.ArgumentException: Value cannot be empty.
Parameter name: data
   at ImageMagick.Throw.IfNullOrEmpty(String paramName, Array value)
   at ImageMagick.MagickImageCollection.Read(Byte[] data, MagickReadSettings readSettings)
   at ImageMagick.MagickImageCollection..ctor(Byte[] data, MagickReadSettings readSettings)
   at ImageToPngService.Service.ImageConverterService.<ConvertFileToPNG>d__0.MoveNext() in C:\\Builds\\16\\CSWeb CI AD Main\\Sources\\ImageToPngService\\Service\\ImageConverterService.cs:line 28

on line  using (MagickImageCollection magickImages = new MagickImageCollection(originalImage, settings))

class methoids
`public IEnumerable<byte[]> ConvertFileToPNG(byte[] originalImage)
        {
            MagickReadSettings settings = new MagickReadSettings()
            {
                Density = new Density(200, 200)
                
            };

            var pngStream = new MemoryStream();

            var convertedImages = new List<byte[]>();

            using (MagickImageCollection magickImages = new MagickImageCollection(originalImage, settings))
            {
                foreach (var image in magickImages)
                {
                    ProcessImage(pngStream, image);
                    yield return pngStream.ToArray();
                }
            }
        }

        private static void ProcessImage(MemoryStream pngStream, IMagickImage image)
        {
            pngStream.SetLength(0);
            image.FilterType = FilterType.Triangle;
            image.UnsharpMask(0.25, 0.25, 8, 0.065);
            image.Quality = 82;
            image.Interlace = Interlace.NoInterlace;
            image.ColorSpace = ColorSpace.sRGB;
            image.Strip();

            image.Write(pngStream, MagickFormat.Png8);
            image.Dispose();
        }`

**Comment by dlemstra** [comment]
You are getting this error because `originalImage` seems to be empty. The length of the array appears to be `0`. You probably want to check that before you pass the data to the `MagickImageCollection`.

**Reply by hammersxo** [reply]
> You are getting this error because the `original image` seems to be empty. The length of the array appears to be `0`. You probably want to [roadrunner email](https://roadrunneremails.net) check that before you pass the data to the `MagickImageCollection`.
Thanks for the suggestion. Will update it later.
